src/agni/client/communication/MessageSender.java

```diff
@@ -1,12 +1,18 @@
 package agni.client.communication;
 
+import java.net.Socket;
+
 public class MessageSender {
 	
-	String ip;
-	int port;
+	private String ip;
+	private int port;
+	private Socket tcpSocket; 
```

Comment:
I think the message sender shouldn't need the ip and port and private members. It only needs that information when connecting the tcp socket. I think we'll need a connect method on the message sender  to connect the tcpSocket with a given `ip` and `port`. The `ip` should be an `InetAddress` and the `port` an unsigned int. 

In terms of moving forward, I think don't need include these two private members then we can add the connect method in another PR.